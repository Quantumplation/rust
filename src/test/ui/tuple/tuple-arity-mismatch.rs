// Issue #6155

fn first((value, _): (isize, f64)) -> isize { value }

fn main() {
    let y = first ((1,2.0,3));
    //~^ ERROR arguments to this function are incorrect
    //~| expected tuple `(isize, f64)`
    //~| found tuple `(isize, f64, {integer})`
    //~| expected a tuple with 2 elements, found one with 3 elements

    let y = first ((1,));
    //~^ ERROR arguments to this function are incorrect
    //~| expected tuple `(isize, f64)`
    //~| found tuple `(isize,)`
    //~| expected a tuple with 2 elements, found one with 1 element
}
