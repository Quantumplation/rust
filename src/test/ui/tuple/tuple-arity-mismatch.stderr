error[E0308]: arguments to this function are incorrect
  --> $DIR/tuple-arity-mismatch.rs:6:13
   |
LL |     let y = first ((1,2.0,3));
   |             ^^^^^^^---------^
   |                    |
   |                    expected `(isize, f64)`, found `(isize, f64, {integer})`
   |
note: function defined here
  --> $DIR/tuple-arity-mismatch.rs:3:4
   |
LL | fn first((value, _): (isize, f64)) -> isize { value }
   |    ^^^^^ ------------------------
help: provide an argument of the correct type
   |
LL |     let y = first({(isize, f64)});
   |             ^^^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/tuple-arity-mismatch.rs:12:13
   |
LL |     let y = first ((1,));
   |             ^^^^^^^----^
   |                    |
   |                    expected `(isize, f64)`, found `(isize,)`
   |
note: function defined here
  --> $DIR/tuple-arity-mismatch.rs:3:4
   |
LL | fn first((value, _): (isize, f64)) -> isize { value }
   |    ^^^^^ ------------------------
help: provide an argument of the correct type
   |
LL |     let y = first({(isize, f64)});
   |             ^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
